ledger/src/blockstore.rs

```diff
@@ -547,11 +559,15 @@ impl Blockstore {
     }
 
     /// Deletes the blockstore at the specified path.
+    ///
+    /// Note that if the `ledger_path` has multiple rocksdb instances, this
+    /// function will destroy all.
     pub fn destroy(ledger_path: &Path) -> Result<()> {
         // Database::destroy() fails if the path doesn't exist
         fs::create_dir_all(ledger_path)?;
-        let blockstore_path = ledger_path.join(BLOCKSTORE_DIRECTORY);
-        Database::destroy(&blockstore_path)
+        Database::destroy(&Path::new(ledger_path).join(BLOCKSTORE_DIRECTORY_ROCKS_LEVEL)).and(
+            Database::destroy(&Path::new(ledger_path).join(BLOCKSTORE_DIRECTORY_ROCKS_FIFO)),
```

Comment:
Want to draw some discussion here:

Are we okay with deleting all the rocksdb instances under a ledger path when `blockstore::destroy` is called?

If the answer is no, then we will need to pass an additional argument to this function to specify which one to delete.

Reply:
I think we'd be ok with deleting both. The separate directory is to prevent mix-and-match; if one was going to call `destroy()`, I think it is reasonable to expect it would wipe everything / both as the previous usage of `destroy()` would do the same.

Per the comment about `destroy()` failing if the path doesn't exist, can you confirm that only having one of [`rocksdb`, `rocksdb_fifo`] doesn't cause this function to throw a bad result?

FWIW, I think lots of validators just do a `rm -rf rocksdb` to remove instead of calling `destroy()`; it looks like the below (which gets called in several places) is the only non-test/bench caller of this function:
https://github.com/solana-labs/solana/blob/911c5a836278470db27e92d79ac840826d17e442/ledger/src/blockstore.rs#L3777-L3783

Reply:
> Per the comment about destroy() failing if the path doesn't exist, can you confirm that only having one of [rocksdb, rocksdb_fifo] doesn't cause this function to throw a bad result?

Confirmed that only one of them exists doesn't cause the function throws by running the ledger_cleanup_test benchmark with `CLEANUP_BLOCKSTORE=true`.

The comment is about the root directory.   Improved the comment to avoid confusion.